src/MSALWrapper/AuthFlow/IntegratedWindowsAuthentication.cs

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.MSALWrapper.AuthFlow
+{
+#if NET472
+    using Microsoft.Identity.Client.Desktop;
+#endif
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.Logging;
+    using Microsoft.Identity.Client;
+
+    /// <summary>
+    /// The broker auth flow.
+    /// </summary>
+    public class IntegratedWindowsAuthentication : IAuthFlow
+    {
+        private readonly ILogger logger;
+        private readonly IEnumerable<string> scopes;
+        private readonly string preferredDomain;
+        private readonly IList<Exception> errors;
+        private IPCAWrapper pcaWrapper;
+
+        #region Public configurable properties
+
+        /// <summary>
+        /// The silent auth timeout.
+        /// </summary>
+        private TimeSpan silentAuthTimeout = TimeSpan.FromMinutes(5);
+
+        /// <summary>
+        /// The intercative windows auth flow timeout.
+        /// </summary>
+        private TimeSpan integratedWindowsAuthTimeout = TimeSpan.FromMinutes(15);

```

Comment:
Since IWA has no interactive path - we should only have 1 Timeout defined. Probably this one, so probably make this one 6 seconds and use it for both?

Reply:
I made the silent timeout (that uses cached accounts) and IWA timeout to be 6 seconds. I am leaving them separate as they still are trying different APIs behind the scenes and this makes it more explicit.

Reply:
integratedWindowsAuthTimeout is now 6 seconds and used in both places.